src/NCronJob/Registry/JobDefinition.cs

```diff
@@ -33,29 +61,38 @@ internal sealed record JobDefinition(
     /// </summary>
     public string? UserDefinedCronExpression { get; set; }
 
-    public object? Parameter { get; set; } = Parameter;
+    public object? Parameter { get; set; }
 
-    public TimeZoneInfo? TimeZone { get; set; } = TimeZone;
+    public TimeZoneInfo? TimeZone { get; set; }
 
     /// <summary>
     /// The JobFullName is used as a unique identifier for the job type including anonymous jobs. This helps with concurrency management.
     /// </summary>
-    public string JobFullName => JobName == Type.Name
-        ? Type.FullName!
-        : $"{typeof(DynamicJobFactory).Namespace}.{JobName}";
+    public string JobFullName { get; }
 
-    private JobExecutionAttributes JobPolicyMetadata { get; } = JobPolicyMetadata ?? new JobExecutionAttributes(Type);
+    private JobExecutionAttributes JobPolicyMetadata { get; }
     public RetryPolicyBaseAttribute? RetryPolicy => JobPolicyMetadata.RetryPolicy;
     public SupportsConcurrencyAttribute? ConcurrencyPolicy => JobPolicyMetadata.ConcurrencyPolicy;
 
     public Type? ExposedType => IsTypedJob ? Type : null;
 
-    public bool IsTypedJob => Type != typeof(DynamicJobFactory);
+    [MemberNotNullWhen(true, nameof(ExposedType))]
+    public bool IsTypedJob { get; }
 
     private bool IsDisabled => CronExpression == NotReacheableCronDefinition;
 
     public bool IsEnabled => CronExpression is null || !IsDisabled;
 
+    public static JobDefinition CreateTyped(
+        Type type,
+        object? parameter)
+    => new(type, parameter);
+
+    public static JobDefinition CreateDynamic(
```

Comment:
`CreateDynamic` doesn't sit 100% right. Because we never used that term for the minimal jobs. The obvious candidate would be `CreateUntyped`

Reply:
Fixed